modules/derivatives_for_hats/derivatives_for_hats.module

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/**
+ * @file
+ * Derivatives for hats!
+ *
+ * Just a note-y note: this doesn't depend on objs_for_hats as OBJ datastreams
+ * could be generated via another method; it just checks if the datastream
+ * exists.
+ */
+
+/**
+ * Implements hook_ten_million_with_a_hat_also_do_these_things().
+ */
+function derivatives_for_hats_ten_million_with_a_hat_also_do_these_things() {
+  return array(
+    array(
+      'callback' => 'derivatives_for_hats_generate_derivatives',
+      // OBJs for hats is weighted at 50.
+      'weight' => 51,
+      'when' => 'between_ingests',
+    ),
+  );
+}
+
+/**
+ * Attempts to generate derivatives from the object OBJ datastream.
+ *
+ * @param AbstractObject $object
+ *   The object to derive for.
+ */
+function derivatives_for_hats_generate_derivatives(AbstractObject $object) {
+  if ($object['OBJ']) {
+    // Recreates islandora_do_batch_derivatives() in immediate mode.
```

Comment:
This is great you should say why in a README, cause if I saw this and didn't know I  would just ctr-x